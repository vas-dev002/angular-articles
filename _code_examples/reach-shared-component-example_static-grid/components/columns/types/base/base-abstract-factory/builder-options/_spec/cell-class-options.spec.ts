import { IColumnMetaSG } from "src/app/components/shared/aggrid/static-grid";
import { CellClassOptions } from "..";

declare const expect: jest.Expect;

describe("[Option]: CellClassOptions", () => {
	const meta = {
		field: "testField",
		cellClass: "testCellClass",
		headerName: "testHeaderName",
		hide: false,
		tooltipField: "testTooltipField",
		lockVisible: false,
		cssClass: "testSlotClass"
	} as unknown as IColumnMetaSG;
	let cellClassOptions: CellClassOptions;

	beforeEach(() => {
		cellClassOptions = new CellClassOptions(meta);
	});

	describe("[CellClassOptions instantiate]", () => {
		it("should be successfully created", () => {
			expect(cellClassOptions).toBeTruthy();
			expect(cellClassOptions).toMatchSnapshot();
		});
	});
});
