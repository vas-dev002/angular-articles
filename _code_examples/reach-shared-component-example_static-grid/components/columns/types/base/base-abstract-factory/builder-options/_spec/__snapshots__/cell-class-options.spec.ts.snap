// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Option]: CellClassOptions [CellClassOptions instantiate] should be successfully created 1`] = `
CellClassOptions {
  "cellClass": "testSlotClass",
  "cellClassRules": Object {
    "sg-cell-disabled": [Function],
  },
}
`;
